Realm/ObjectServerTests/RLMAsymmetricSyncServerTests.mm

```diff
@@ -0,0 +1,322 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import "RLMSyncTestCase.h"
+#import "RLMObject_Private.h"
+#import "RLMApp_Private.h"
+#import "RLMObjectSchema_Private.h"
+
+// These are defined in Swift. Importing the auto-generated header doesn't work
+// when building with SPM, so just redeclare the bits we need.
+@interface RealmServer : NSObject
```

Comment:
`RLMFlexibleSyncTests` also defines an interface for `RealmServer`. I think we should create a helper header file that defines this interface so that both test suites can use it, and removes the need to repeat the interface in both places.

Reply:
I just centralised everything in `RLMSyncTestCases.h`https://github.com/realm/realm-swift/blob/0b3b8720e0de3ab2561d585e126f6549c006e079/Realm/ObjectServerTests/RLMSyncTestCase.h#L28  file which is added to all sync test files.